### ISSUE TITLE
`c8y_log_plugin` doesn't receive new messages after `mosquitto` restarts

### ISSUE THREAD

**Issue by rina23q** [thin-edge/thin-edge.io#1286]
**Describe the bug**
`c8y_log_plugin` doesn't receive new MQTT messages after `mosquitto` restarts during the runtime of `c8y_log_plugin`.

**To Reproduce**
Copied from [here](https://github.com/thin-edge/thin-edge.io/issues/1184#issuecomment-1185533832).

[Precondition]
- All thin-edge packages are installed in the device.
- The device is already connected to c8y by `sudo tedge connect c8y`.
- The c8y_log_plugin process is up. Run the binary directly (execute `c8y_log_plugin --debug` in the shell). 

[Steps]
1. Publish a log request MQTT message. 
```
tedge mqtt pub 'c8y/s/ds' '522,DeviceSerial,logfileA,2013-06-22T17:03:14.000+02:00,2013-06-22T18:03:14.000+02:00,ERROR,1000'
```
2. Check the output of `c8y_log_plugin`. You can see that the plugin received the message by 1).
```
2022-07-15T13:08:08.564046292Z  INFO c8y_log_plugin: Log request received: 522,DeviceSerial,logfileA,2013-06-22T17:03:14.000+02:00,2013-06-22T18:03:14.000+02:00,ERROR,1000
```
3. Stop `mosquitto` service.
```
sudo systemctl stop mosquitto
```
4. Start `mosquitto` service.
```
sudo systemctl start mosquitto
```
5. Publish the same message as 1).
```
tedge mqtt pub 'c8y/s/ds' '522,DeviceSerial,logfileA,2013-06-22T17:03:14.000+02:00,2013-06-22T18:03:14.000+02:00,ERROR,1000'
```
6. Check the output of `c8y_log_plugin`. You can't see that the plugin received the new message by 5). 

**Expected behavior**
After restarting `mosquitto`, still `c8y_log_plugin` can receive MQTT messages published by anywhere onto the topics `c8y/s/ds` or `tedge/health/mosquitto-c8y-bridge`.

**Environment (please complete the following information):**
 - OS [Ubuntu 20.04]
 - Hardware [Laptop]
 - System-Architecture [amd64]
 - thin-edge.io version [0.7.2]

**Additional context**
This bug might be the same as reported in https://github.com/thin-edge/thin-edge.io/issues/1194.

`tedge_mapper c8y` is actually using the same MQTT crates, however, it doesn't have any issue after restarting `mosquitto`. I think the issue is mqtt client implementation of `c8y_log_plugin`.

**Comment by rina23q** [comment]
And another test case provided by @initard [here](https://github.com/thin-edge/thin-edge.io/issues/1184#issuecomment-1191338668) is also valid.

Test 2: When bridge is down
Given that the DUT is NOT connected to c8y

sudo tedge disconnect c8y
And the c8y_log_plugin is started

sudo systemctl start c8y-log-plugin.service
When DUT is connected to c8y

sudo tedge connect c8y
Then c8y supported log types are uploaded to c8y

cloud-side

**Comment by PradeepKiruvale** [comment]
The issue is related to the re-connection of mqtt client after mosquito reboots.
In the receiver loop of the mqtt client when the client disconnects and reconnects the resubscription has to be done.
`

                Ok(Event::Incoming(Packet::ConnAck(ack))) => {
                       eprintln!("INFO: Connected");           
                       let subscriptions = config.subscriptions.filters();
                       if subscriptions.is_empty() {
                          break;
                       }
                      mqtt_client.subscribe_many(subscriptions).await.unwrap();                   
                }
`

**Comment by PradeepKiruvale** [comment]
steps to test this

Install the thin-edge from here: https://github.com/thin-edge/thin-edge.io/actions/runs/3015303111
1. subscribe to `c8y/s/us` using `sudo tedge mqtt sub 'c8y/s/us'`
2. `sudo systemctl start c8y-log-plugin.service`
3. Wait for some time
4. `sudo tedge connect c8y` 
5. The subscriber that started in the step 1, should  receive `118,software-management`

**Comment by PradeepKiruvale** [comment]
Resolved with #1360
